documentation/src/main/wiki/Splitter.md

```diff
@@ -1,4 +1,35 @@
-# TODO
-- what is splitter - input and output
-- when it works
-- how it works
+#HTML Fragment Splitter
+HTML Fragment Splitter divides HTML template into static and dynamic chunks. Those chunks (fragments) 
+goes to Knot context and can be processed later by [Knots](#Knot).
+
+##How does it work?
+HTML Fragment Splitter gets a Knot context as input and responds with the modified Knot context. It 
+divides HTML using regexp `<script\s+data-api-type\s*=\s*"([A-Za-z0-9-]+)"[^>]*>.+?</script>` into
+static and dynamic fragments. So all `script` tags with a `data-api-type` attribute are converted to 
+dynamic fragments. <b>According to performance reasons Splitter requires `data-api-type` 
+attribute to be the first attribute in the `script` tag.</b>
+All HTML markup outside script tags is converted static fragments.
+
+Fragment contains an identifier, a content (a template chunk) and a context. The <b>identifier</b> has `data-api-type` 
+attribute value or `_raw` for static fragments. It can be used by Knots to select required fragment / fragments 
+(performance enhancement) without additional snippet content processing. The <b>content</b> contains 
+script tag with its content for dynamic fragments or static HTML content for static fragments. 
+The context can be omitted at this moment.
+
+At the end Splitter updates the Knot context with the list of fragments and returns it to further processing.
```

Comment:
Please provide example with very simple markup before and after splitting e.g.

### Example

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Knot.x example</title>
</head>
<body>
  <div class="row">
    <script data-api-type="templating"
            data-service="first-service"
            type="text/x-handlebars-template">
      <div class="col-md-4">
        <h2>Snippet - {{_result.message}}</h2>
      </div>
    </script>
  </div>
</body>
</html>
```

will be split into 3 following *fragments* by Splitter:
Fragment 1
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Knot.x example</title>
</head>
<body>
  <div class="row">
```  
Fragment 2 (Snippet)
```html
    <script data-api-type="templating"
            data-service="first-service"
            type="text/x-handlebars-template">
      <div class="col-md-4">
        <h2>Snippet - {{_result.message}}</h2>
      </div>
    </script>
```
Fragment 3
```html
  </div>
</body>
</html>
```